python/ray/services.py

```diff
@@ -709,9 +710,31 @@ def start_local_scheduler(redis_address,
         # By default, use the number of hardware execution threads for the
         # number of cores.
         resources["CPU"] = psutil.cpu_count()
+
+    # See if CUDA_VISIBLE_DEVICES has already been set.
+    gpu_ids_str = os.environ.get("CUDA_VISIBLE_DEVICES", None)
+    if gpu_ids_str is None:
+        gpu_ids = None
+    elif gpu_ids_str == "":
+        gpu_ids = []
+    else:
+        gpu_ids = [int(i) for i in gpu_ids_str.split(",")]
```

Comment:
Can we separate out the above block into a `utils` function?

Reply:
fixed